src/vm/object.cpp

```diff
@@ -1969,54 +1969,163 @@ StackTraceElement & StackTraceArray::operator[](size_t index)
 }
 
 #if !defined(DACCESS_COMPILE)
-// Define the lock used to access stacktrace from an exception object
+// Define the lock used to access stacktrace from a global exception type
 SpinLock g_StackTraceArrayLock;
 
 void ExceptionObject::SetStackTrace(StackTraceArray const & stackTrace, PTRARRAYREF dynamicMethodArray)
 {        
     CONTRACTL
     {
-        GC_NOTRIGGER;
-        NOTHROW;
+        THROWS;
+        GC_TRIGGERS;
         MODE_COOPERATIVE;
     }
     CONTRACTL_END;
 
-    Thread *m_pThread = GetThread();
-    SpinLock::AcquireLock(&g_StackTraceArrayLock, SPINLOCK_THREAD_PARAM_ONLY_IN_SOME_BUILDS);
+    if (GetMethodTable() == g_pOutOfMemoryExceptionClass)
+    {
+        // We can't activate the SyncBlock for an OutOfMemoryException, so use preallocated SpinLock
+        Thread* m_pThread = GetThread();
```

Comment:
This `Thread* m_pThread = GetThread();` can be deleted. It does not do anything useful.

Reply:
Removed